sql/catalyst/src/test/scala/org/apache/spark/sql/HashBenchmark.scala

```diff
@@ -119,11 +146,12 @@ object HashBenchmark {
       .add("map", mapOfInt)
       .add("mapOfMap", MapType(IntegerType, mapOfInt))
     /*
-    Intel(R) Core(TM) i7-4960HQ CPU @ 2.60GHz
+    Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
     Hash For map:                       Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
     -------------------------------------------------------------------------------------------
-    interpreted version                      1820 / 1861          0.0      444347.2       1.0X
-    codegen version                           205 /  223          0.0       49936.5       8.9X
+    interpreted version                      1612 / 1618          0.0      393553.4       1.0X
+    codegen version                           149 /  150          0.0       36381.2      10.8X
+    codegen version 64-bit                    144 /  145          0.0       35122.1      11.2X
      */
```

Comment:
This result is because we have removed the `hashCode` implementation from `ArrayBasedMapData`, and the system default `hashCode` is really fast(almost no-op).

We did this because they have same logic at the time, but it's not true now. @gatorsmile do you wanna send a PR to fix this(use interpreted hash expression instead of `BaseGenericInternalRow.hashCode`)? thanks!